### ISSUE TITLE
`workspacePrompt` doesn't seem to work with fuzzy matching

### ISSUE THREAD

**Issue by pbogdan** [xmonad/xmonad-contrib#366]
### Problem Description

It seems I can't make `workspacePrompt` working with fuzzy matching.

Upon bringing up the prompt I expect it to behave as described in the `XMonad.Prompt.FuzzyMatch` module, instead a match is not found unless the entered text is a prefix of the workspace's name.

### Configuration File

N/A

The relevant parts are:

Prompt config:

```haskell
myXPConfig :: XPConfig
myXPConfig = def { font            = "xft:PRagmataPro:pixelsize=32"
                 , height          = 39
                 , bgColor         = black
                 , fgColor         = white
                 , searchPredicate = fuzzyMatch
                 , sorter          = fuzzySort
                 , alwaysHighlight = True
                 }
```
Bindings:

```haskell
       , ("M-S-g", workspacePrompt myXPConfig (windows . W.greedyView))
       , ( "M-S-f"
         , windowPrompt myXPConfig Goto allWindows
         )
```

I included the `windowPrompt` one as I confirmed that one behaves as expected.

When I have I will try and create a minimal reproducible example. Please feel free to close this until a minimal example is provided if that's a hard requirement.

### Checklist

  - [ ] I've read [CONTRIBUTING.md](https://github.com/xmonad/xmonad/blob/master/CONTRIBUTING.md)

  - [ ] I tested my configuration with [xmonad-testing](https://github.com/xmonad/xmonad-testing)

**Comment by pbogdan** [comment]
I reduced my config to:

```haskell
module Main where

import           Data.Map                       ( Map )
import           System.Exit
import           XMonad
import           XMonad.Prompt.Workspace
import           XMonad.Actions.DynamicWorkspaces
import           XMonad.Prompt.FuzzyMatch
import           XMonad.Config.Desktop
import           XMonad.Hooks.EwmhDesktops
import           XMonad.Prompt
import qualified XMonad.StackSet               as W
import           XMonad.Util.EZConfig

myWorkspaces :: [String]
myWorkspaces = map show ([1 .. 5] :: [Integer]) ++ [ "foobar" ]

myXPConfig :: XPConfig
myXPConfig = def { font            = "xft:PRagmataPro:pixelsize=32"
                 , height          = 39
                 , bgColor         = black
                 , fgColor         = white
                 , searchPredicate = fuzzyMatch
                 , sorter          = fuzzySort
                 }

myKeymap :: XConfig Layout -> Map (KeyMask, KeySym) (X ())
myKeymap c =
  mkKeymap c
    $  [
        ("M-S-c"    , kill)
       , ("M-S-n", appendWorkspacePrompt myXPConfig)
       , ("M-c"  , removeEmptyWorkspace)
       , ("M-S-g", workspacePrompt myXPConfig (windows . W.greedyView))

       , ("M-<Tab>"   , windows W.focusDown)
       , ("M-j"       , windows W.focusDown)
       , ("M-k"       , windows W.focusUp)
       , ("M-m"       , windows W.focusMaster)
       , ("M-<Return>", windows W.swapMaster)
       , ("M-S-j"     , windows W.swapDown)
       , ("M-S-k"     , windows W.swapUp)
       , ("M-S-q"     , io exitSuccess)
       , ( "M-q"
         , restart "xmonadrc" True
         )
       ]
    ++ [ (m ++ i, windows $ f j)
       | (i, j) <- zip (map show ([1 .. 9] :: [Integer])) (XMonad.workspaces c)
       , (m, f) <- [("M-", W.greedyView), ("M-S-", W.shift)]
       ]
    ++ [ (m ++ key, screenWorkspace sc >>= flip whenJust (windows . f))
       | (key, sc) <- zip ["w", "e", "r"] [0 ..]
       , (m  , f ) <- [("M-", W.view), ("M-S-", W.shift)]
       ]

main :: IO ()
main = launch $ ewmh defaults

white :: String
white = "#fbf8ef"

black :: String
black = "#383a42"

defaults =
  desktopConfig
        {
         workspaces = myWorkspaces
         , keys               =  myKeymap
        }
```

and the issue is still reproducible.

**Comment by pbogdan** [comment]
OK, NVM, this has been fixed in 28b3e34fd77c19e6b935c01a40e66cbf90abf69a Sorry for the noise.
